service/worker/scheduler/workflow.go

```diff
@@ -246,12 +250,39 @@ func (s *scheduler) run() error {
 		for s.processBuffer() {
 		}
 		s.updateMemoAndSearchAttributes()
+
+		// if schedule is not paused and out of actions or do not have anything scheduled, delete the schedule workflow after retention period is passed
+		if s.tweakables.RetentionTime > 0 && (!s.Schedule.State.Paused && (nextWakeup.IsZero() || !s.canTakeScheduledAction(false, false))) {
+			// if schedule is empty use the previousProcessedTime as last workflow start time
```

Comment:
1. hmmm, I don't remember now why I did not use `t1` 🤔 it does look it will work. Let me test it. 
2. I don't think we need to have these two times to agree. As long as the duration we use in `sleep` falls within the duration we check here, we should be good. The order is: `previousProcessedTime` -> `RecentActions[-1].time`->`s.now()` inside sleep loop ---[sleep time]---> `s.now()` inside `run` loop. 
3. by `schedule` do you mean the schedule workflow? which parameter can I use for this?

Reply:
I see, yeah that makes perfect sense, and very good point on update without action not resetting the timer. Let me update the code then.